dashboard/lib/services/teacher.rb

```diff
@@ -0,0 +1,15 @@
+module Services
+  module Teacher
+    # Updates the user permission to AUTHORIZED_TEACHER
+    #
+    # @param [User]
+    def self.verify(teacher)
+      return false unless teacher.teacher?
+      return true if teacher.verified_teacher?
```

Comment:
After some consideration, I think we should remove this check and call it outside of the method if necessary.
Here are a few reasons for this decision:

1. `#verified_teacher?` will return `true` even if the permission is not saved.
2. `#save` will not commit changes to the database if the record hasn’t changed anyway (if the user is already verified).
3. This approach will allow us to re-verify the user in the future if needed.
```suggestion
```

Reply:
> #save will not commit changes to the database if the record hasn’t changed anyway (if the user is already verified).
Will save return true in that case?

Reply:
Yes, it will

Reply:
Assigning the permissions won't have an effect if the permissions already exist:
https://github.com/code-dot-org/code-dot-org/blob/ee8124ab49e69f35c9941b87526a84f1c3a7b6c8/dashboard/app/models/concerns/user_permission_grantee.rb#L28-L32

Reply:
We had an issue where we were adding the verified teacher permission to already verified teachers and it was sending additional emails. How was that triggering?

Reply:
I just remember the PR: https://github.com/code-dot-org/code-dot-org/pull/60539

Reply:
I added the additional check as in that PR just to be sure.